src/DIPS.Xamarin.UI/Controls/Popup/PopupBehaviour.cs

```diff
@@ -0,0 +1,127 @@
+﻿using System;
+using System.Windows.Input;
+using DIPS.Xamarin.UI.Extensions;
+using Xamarin.Forms;
+
+namespace DIPS.Xamarin.UI.Controls.Popup
+{
+    public class PopupBehaviour : Behavior<View>
+    {
+        private readonly Command m_onTappedCommand;
+        private View? m_attachedTo;
+        public PopupBehaviour()
+        {
+            m_onTappedCommand = new Command(ShowPopup);
+        }
+
+        protected override void OnAttachedTo(View bindable)
+        {
+            m_attachedTo = bindable;
+            if (bindable is Button button)
+            {
+                button.Command = m_onTappedCommand;
+            }
+            else
+            {
+                bindable.GestureRecognizers.Add(new TapGestureRecognizer { Command = m_onTappedCommand });
+            }
+
+            base.OnAttachedTo(bindable);
+        }
+
+        protected override void OnDetachingFrom(View bindable)
+        {
+            base.OnDetachingFrom(bindable);
+        }
+
+        public static readonly BindableProperty CloseOnClickProperty =
```

Comment:
This property is also now visible on the `PopupBehaviour` (which is kind of strange). Can we work around this to not make it visible there, even though we are going to use it in the behaviour.

Reply:
I think this can potentially exist in a different class, to not make it visible.